game-core/src/main/java/games/strategy/triplea/ui/screen/TileManager.java

```diff
@@ -336,24 +339,27 @@ private static void drawTerritoryEffects(final Territory territory, final MapDat
   }
 
   private void drawUnits(final Territory territory, final MapData mapData, final Set<Tile> drawnOn,
-      final Set<IDrawable> drawing) {
+      final Set<IDrawable> drawing, final GameData data) {
     final Iterator<Point> placementPoints = mapData.getPlacementPoints(territory).iterator();
     if (placementPoints == null || !placementPoints.hasNext()) {
       throw new IllegalStateException("No where to place units:" + territory.getName());
     }
+
     Point lastPlace = null;
-    final Iterator<UnitCategory> unitCategoryIter =
-        UnitSeperator.categorize(territory.getUnits().getUnits()).iterator();
-    while (unitCategoryIter.hasNext()) {
-      final UnitCategory category = unitCategoryIter.next();
+    final List<UnitCategory> categories = getSortedUnitCategories(territory, data);
+    for (final UnitCategory category : categories) {
```

Comment:
There's no need to explicitly declare a variable for the categories

Reply:
Updated.